### PR TITLE
Add support for Namecheap

### PR DESCRIPTION
**Pull request by pschmitt**
This leverages the [PyNamecheap](https://github.com/Bemmu/PyNamecheap) library and provides Namecheap support

### CODE REVIEW

**Comment by CapsuleCD** [comment]
Hi.

I'm an automated pull request bot named [CapsuleCD](https://www.capsulecd.com). I handle testing, versioning and package releases for this project. 
- If you're the owner of this repo, you can click the button below to kick off the CapsuleCD build pipeline and create an automated release.'
- If not, don't worry, someone will be by shortly to check on your pull request. 

[CapsuleCD](https://www.capsulecd.com/project/github/AnalogJ/lexicon/pullrequests/135)

---
If you're interested in learning more about [CapsuleCD](http://www.github.com/AnalogJ/capsulecd), or adding it to your project, you can check it out [here](https://www.capsulecd.com)

File: optional-requirements.txt [hunk 1]
```diff
@@ -15,5 +15,6 @@
 ###############################################################################
 
 --process-dependency-links
+.[Pynamecheap]
```
Comment (AnalogJ) [inline optional-requirements.txt]: this should be the same as the key in `extras_require`, ie. `.[namecheap]`

**Comment by AnalogJ** [comment]
Hey @pschmitt 
Is there anything I can do to help get this PR merged?
Did you get a chance to create the VCR recordings?

**Comment by pschmitt** [comment]
Oh yeah I forgot about the recordings. I'll what I can do about that today.

**Comment by pschmitt** [comment]
So I'm issuing the following commands, but none of the env vars get passed through. 
Am I missing something?

```bash
export LEXICON_NAMECHEAP_USERNAME="MY_USERNAME"
export LEXICON_NAMECHEAP_AUTH_TOKEN="MY_AUTH_TOKEN"
py.test tests/providers/test_namecheap.py
```

`tests/providers/test_namecheap.py`:

```python
# Test for one implementation of the interface
from lexicon.providers.namecheap import Provider
from integration_tests import IntegrationTests
from unittest import TestCase
import pytest

# Hook into testing framework by inheriting unittest.TestCase and reuse
# the tests which *each and every* implementation of the interface must
# pass, by inheritance from integration_tests.IntegrationTests
class NamecheapProviderTests(TestCase, IntegrationTests):

    Provider = Provider
    provider_name = 'namecheap'
    domain = 'comreset.io'

    def _filter_post_data_parameters(self):
        return ['login_token']

    def _filter_headers(self):
        return ['Authorization']

    def _filter_query_parameters(self):
        return ['secret_key']

    @pytest.mark.skip(reason="can not set ttl when creating/updating records")
    def test_Provider_when_calling_list_records_after_setting_ttl(self):
        return
```

**Comment by AnalogJ** [comment]
I think that should be:
```
export LEXICON_NAMECHEAP_USERNAME="MY_USERNAME"
export LEXICON_NAMECHEAP_TOKEN="MY_AUTH_TOKEN"
py.test tests/providers/test_namecheap.py
```

**Comment by pschmitt** [comment]
Still, none of the env vars land in the config (I verified by ppriting the config)

**Comment by ppmathis** [comment]
I am working on a ClouDNS integration (pull request follows soon) and had the same problem, but was able to fix it by using the following code snippet:

```python
def _test_options(self):
    cmd_options = super(CloudnsProviderTests, self)._test_options()
    cmd_options.update(env_auth_options(self.provider_name))
    return cmd_options
```

The function **env_auth_options** seems not to be called automatically in the test suite, which leads to the issue that @pschmitt currently has.

**Comment by AnalogJ** [comment]
Hmm. looks like there might need to be some tweaking in the way that we populate the options/config dictionary. 

@pschmitt did the workaround that @ppmathis used work for you? I'd like to get this provider merged first, and then take a look at fixing that auth options merging logic.

**Comment by AnalogJ** [comment]
Hey @pschmitt 
Can you try rebasing your code/merging the latest master branch into your branch? I think I fixed the config options issue, and I'd really like to get your PR merged.

Thanks!

**Comment by AnalogJ** [comment]
closes #145

**Comment by AnalogJ** [comment]
closed by #135 
Released in v2.1.17

**Pull request closed without merging**